rust-toolchain.toml

```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "1.80"
+components = [ "rustfmt" ]
```

Comment:
I think you should rather just [set the minimum supported Rust version in the Cargo.toml](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field) and remove this file.

Reply:
I disagree, if this was a library for others to consume then sure, but since this is an application I think it is good to pin the version like any other dependency.

Reply:
Never versions introduce new clippy warnings, and we don't want the CI to randomly start failing checks because a new stable release of rust.
We can remove this file, but in the end we will pin a rust version in the build job which is the rust version we care about.

Reply:
Pinning in the build job is fine and makes sense - but if someone wants to build this with a newer compiler, I think they should be able to (even if they get some clippy warnings).

Reply:
You can always override using `cargo +stable`